gslab_scons/tests/test_configuration_tests.py

```diff
@@ -0,0 +1,394 @@
+
+        def make_yaml(string):
+            if os.path.isfile('yaml.yaml'):
+                os.remove('yaml.yaml')  
+            with open('yaml.yaml', 'wb') as f:
+                f.write('%s\n' % string)
+
+        # Test Good
+
+        # Yaml file exists, is not corrupt, and has key.
+        make_yaml('key: value')
+        self.assertEqual(configuration_tests.load_yaml_value('yaml.yaml', 'key'), 'value')
+
+        # Yaml file exists, is not corrupt, and doesn't have key. User enters value for key.
+        make_yaml('bad_key: bad_value')
+        self.assertEqual(configuration_tests.load_yaml_value('yaml.yaml', 'key'), 'value')
+        
+        # But now yaml has a correct key and doesn't require user input.
+        mock_raw_input.side_effect = lambda x: "bad_value"
+        self.assertEqual(configuration_tests.load_yaml_value('yaml.yaml', 'key'), 'value')
+        mock_raw_input.side_effect = raw_input_side_effect
+
+        # Yaml file exists and is corrupt. User deletes and re-creates. 
+        make_yaml('key value')
+        self.assertEqual(configuration_tests.load_yaml_value('yaml.yaml', 'key'), 'value')
+
+        # Yaml file does not exist. User enters value to create. 
+        os.remove('yaml.yaml')
+        self.assertEqual(configuration_tests.load_yaml_value('yaml.yaml', 'key'), 'value')
+
+        # Yaml file does not exist. User enters none value to create. 
+        os.remove('yaml.yaml')
+        mock_raw_input.side_effect = raw_input_side_effect2
+        self.assertEqual(configuration_tests.load_yaml_value('yaml.yaml', 'stata_executable'), None)
+
+        # Yaml file now exists with none value.
+        self.assertEqual(configuration_tests.load_yaml_value('yaml.yaml', 'stata_executable'), None)
+
+        # Test bad
+        with self.assertRaises(ex_classes.PrerequisiteError):
+            # Corrupt file and user doesn't choose to delete and re-create 
+            make_yaml('key value')
+            configuration_tests.load_yaml_value('yaml.yaml', 'key')
+
+        if os.path.isfile('yaml.yaml'):
+            os.remove('yaml.yaml')  
+
+
+    @mock.patch('gslab_scons.configuration_tests.is_unix')
+    @mock.patch('gslab_scons.configuration_tests.convert_packages_argument')
+    @mock.patch('gslab_scons.configuration_tests.subprocess.check_output')
+    def test_check_stata_packages(self, mock_check_output, mock_package_argument, 
+                                  mock_is_unix):
+
+        def check_output_side_effect(*args, **kwargs):
+            command = args[0].split("which ")[0]
+            if not re.search("statamp", command):
+                raise subprocess.CalledProcessError(1, args[0])  
+
+            with open('stata.log', 'wb') as f:
+                if not re.search('yaml', args[0]):
+                    package = args[0].split("which ")[1]
+                    f.write("command %s not found" % package)
+
+        def is_unix_side_effect_true():
```

Comment:
Consider `return_value` (I'll stop commenting this now and leave it to you to decide where it applies).